osquery/tables/system/windows/smbios_tables.cpp

```diff
@@ -67,5 +152,69 @@ QueryData genPlatformInfo(QueryContext& context) {
   results.push_back(r);
   return results;
 }
+
+QueryData genMemoryDevices(QueryContext& context) {
+  QueryData results;
+
+  std::string query = "select * from Win32_PhysicalMemory";
+  const auto request = WmiRequest::CreateWmiRequest(query);
+  if (!request || !request->getStatus().ok()) {
+    return results;
+  }
+  const std::vector<WmiResultItem>& wmiResults = request->results();
+  if (wmiResults.size() <= 0) {
+    return results;
+  }
+  for (int i = 0; i < wmiResults.size(); ++i) {
+    Row r;
+    // Unable to find match for these from WMI.
+    r["handle"] = "";
+    r["array_handle"] = "";
+    long formFactorId;
+    wmiResults[i].GetLong("FormFactor", formFactorId);
+    r["form_factor"] = getFormFactor(formFactorId);
+    long totalWidth = 0;
+    wmiResults[i].GetLong("TotalWidth", totalWidth);
+    r["total_width"] = INTEGER(totalWidth);
+    long dataWidth = 0;
+    wmiResults[i].GetLong("DataWidth", dataWidth);
+    r["data_width"] = INTEGER(dataWidth);
+    std::wstring capacityWStr;
+    wmiResults[i].GetString(stringToWstring("Capacity"), capacityWStr);
+    r["size"] = INTEGER(getMemorySize(capacityWStr));
+    // Unable to find match for set in WMI.
+    r["set"] = INTEGER(-1);
```

Comment:
Please group the unfound things together?

Also, I'd probably prefer blank or null, to -1.